lib/hydra/works/models/concerns/generic_file/derivatives.rb

```diff
@@ -5,24 +5,27 @@ module Derivatives
     included do
       include Hydra::Derivatives
 
-      # This was taken directly from Sufia's GenericFile::Derivatives and modified to exclude any processing that modified the original file
+      # Sets output_file_service to PersistDerivative instead of default Hydra::Derivatives::PersistBasicContainedOutputFileService
+      Hydra::Derivatives.output_file_service = Hydra::Works::PersistDerivative
+
       makes_derivatives do |obj|
         case obj.mime_type
         when *pdf_mime_types
-          obj.transform_file :original_file, { thumbnail: { format: 'jpg', size: '338x493', datastream: 'thumbnail' } }, processor: :work
+          obj.transform_file :original_file, thumbnail: { format: 'jpg', size: '338x493' }
         when *office_document_mime_types
-          obj.transform_file :original_file, { thumbnail: { format: 'jpg', size: '200x150>', datastream: 'thumbnail' } }, processor: :document
+          obj.transform_file :original_file, { thumbnail: { format: 'jpg', size: '200x150>' } }, processor: :document
+        when *audio_mime_types
+          obj.transform_file :original_file, { mp3: { format: 'mp3' }, ogg: { format: 'ogg' } }, processor: :audio
         when *video_mime_types
-          obj.transform_file :original_file, { thumbnail: { format: 'jpg', datastream: 'thumbnail' } }, processor: :video
+          obj.transform_file :original_file, { webm: { format: 'webm' }, mp4: { format: 'mp4' }, thumbnail: { format: 'jpg' } }, processor: :video
         when *image_mime_types
-          obj.transform_file :original_file, { thumbnail: { format: 'jpg', size: '200x150>', datastream: 'thumbnail' } }, processor: :work
+          obj.transform_file :original_file, thumbnail: { format: 'jpg', size: '200x150>' }
         end
+        
       end
 
     end
 
-    # This bit was not in Sufia, but needs to be here for derivatives to work
-    # TODO: delegate mime type somewhere else?
```

Comment:
Yes.  I'm looking at having the returned objects from derivatives have a mime_type method.